x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -148,6 +149,30 @@ export function InnerWorkspacePanel({
     ]
   );
 
+  const variables = useMemo(() => {
+    const indexPatternRefs: Record<string, string> = {};
+
+    Object.entries(datasourceMap).forEach(([datasourceId, datasource]) => {
+      const state = datasourceStates[datasourceId].state;
+      datasource
+        .getMetaData(state)
+        .indexPatternsByLayer.forEach(({ layerId, indexPatternId, refName }, index) => {
+          indexPatternRefs[`filterable-index-pattern-${index}`] = indexPatternId;
+          indexPatternRefs[refName] = indexPatternId;
+        });
+    });
+    return indexPatternRefs;
+  }, [datasourceMap, datasourceStates]);
+
+  const context: ExecutionContextSearch = {
+    query: framePublicAPI.query,
+    timeRange: {
+      from: framePublicAPI.dateRange.fromDate,
+      to: framePublicAPI.dateRange.toDate,
+    },
+    filters: framePublicAPI.filters,
+  };
```

Comment:
This is a feature we should have been using a long time ago, it simplifies the code a lot.